caliperreader: handle root nodes in _create_parent

slabasan opened the pull request:
When creating parent nodes, we need to handle the case that the parent might be a root node. Previously, the recursive _create_parent calls were being made on root nodes, and we incorrectly tried to index into the grandparent callpath tuple, even though it was empty. This ends the recursion if we encounter an empty callpath tuple.

tgamblin commented:
@slabasan: can you put some context for why we're making this change in the PR?  What wasn't happening, why do we need to sort the index, etc?

bhatele commented:
Its still unclear why do we need to sort the data frame? Do certain functions in the hatchet API assume a sorted dataframe?

slabasan commented:
If we do a column-wise subtract of the dataframe, I think the indexes need to align.

slabasan commented:
Closing this PR in favor of #75.

slabasan committed:
caliperreader: handle root nodes in _create_parent

When creating parent nodes, we need to handle the case that the parent might be
a root node. Previously, the recursive _create_parent calls were being made on
root nodes, and we incorrectly tried to index into the grandparent callpath
tuple, even though it was empty. This ends the recursion if we encounter an
empty callpath tuple.
